core/misc/tabledrag.js

```diff
@@ -633,6 +633,12 @@ Backdrop.tableDrag.prototype.pointerCoords = function (event) {
     return { x: event.pageX, y: event.pageY };
   }
 
+  // If pageX and pageY are defined in the event.originalEvent, use them instead
+  // of clientX and clientY to avoid potentially faulty pointer position calculations 
+  if (event.originalEvent.pageX || event.originalEvent.pageY) {
```

Comment:
Ah yes my comment wasn't accurate, thanks. If pageX OR pageY are defined --- use them.

Reply:
Maybe the comment is right, and the code needs an update? :grinning:

You actually access both properties, so an "and" in the if clause might make sense. (I know, that's a bit pedantic...)

Reply:
Oh, I see what you're saying.

Changing to an AND in the code would break the behavior if your pointer position was zero on the X axis, so I don't think I'd want to do that? A zero is a legitimate coordinate although it equates to false, and we'd still want the Y position returned. I can imagine that while you're dragging table rows around you may even drag your cursor all the way to the left side of the screen, resulting in a "0".

The original broken behavior was that the code was returning the position within the viewport (clientX / clientY) instead of the document position when tabledrag was trying to calculate whether and how fast to scroll the screen, resulting in broken / jerky scrolling that interrupted the drag & drop.

I think originally I was just assuming whoever wrote the first "if (event.pageX || event.pageY) {" knew what they were doing. Thoughts?

Reply:
> Changing to an AND in the code would break the behavior if your pointer position was zero on the X axis

Ah, now I get it, this can be zero, of course. And if it's not set – is it "undefined" then?